File: frontend/src/api/auditLogs.ts
import { apiClient } from './client'
import type { AuditLogListResponse, AuditLogQueryParams } from '../types/auditLog'

export const auditLogsApi = {
  getLogs: (params?: AuditLogQueryParams) => {
    const q = new URLSearchParams()
    if (params?.page !== undefined) q.set('page', String(params.page))
    if (params?.page_size !== undefined) q.set('page_size', String(params.page_size))
    if (params?.action) q.set('action', params.action)
    if (params?.actor_id !== undefined) q.set('actor_id', String(params.actor_id))
    if (params?.target_type) q.set('target_type', params.target_type)
    if (params?.is_success !== undefined) q.set('is_success', String(params.is_success))
    if (params?.from_date) q.set('from_date', params.from_date)
    if (params?.to_date) q.set('to_date', params.to_date)
    if (params?.search) q.set('search', params.search)

    const qs = q.toString()
    return apiClient.get<AuditLogListResponse>(`/audit-logs${qs ? `?${qs}` : ''}`)
  },
}

File: backend/app/api/routes/audit_logs.py
from datetime import datetime
from typing import Optional

from fastapi import APIRouter, Depends, Query, status
from sqlalchemy.orm import Session

from app.api.deps import get_current_user
from app.core.database import get_db
from app.models.user import User
from app.schemas.audit_log import AuditLogListResponse
from app.services import audit_log_service as service

router = APIRouter(
    prefix="/audit-logs",
    tags=["audit-logs"],
    dependencies=[Depends(get_current_user)],
)


@router.get("", response_model=AuditLogListResponse, status_code=status.HTTP_200_OK)
def get_audit_logs(
    page: int = Query(1, ge=1),
    page_size: int = Query(50, ge=1, le=200),
    action: Optional[str] = Query(None),
    actor_id: Optional[int] = Query(None),
    target_type: Optional[str] = Query(None),
    is_success: Optional[bool] = Query(None),
    from_date: Optional[datetime] = Query(None),
    to_date: Optional[datetime] = Query(None),
    search: Optional[str] = Query(None),
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    return service.get_audit_logs(
        db,
        current_user,
        page=page,
        page_size=page_size,
        action=action,
        actor_id=actor_id,
        target_type=target_type,
        is_success=is_success,
        from_date=from_date,
        to_date=to_date,
        search=search,
    )
